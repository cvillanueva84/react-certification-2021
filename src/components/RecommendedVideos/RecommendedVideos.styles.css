.recommended-video__card {
  width: 100%;
  display: flex;
}

.recommended-video__card-img {
  width: 40%;
  margin-left: 2rem;
}

.recommended-video__card-text {
  width: 60%;
  display: flex;
  align-items: center;
}
