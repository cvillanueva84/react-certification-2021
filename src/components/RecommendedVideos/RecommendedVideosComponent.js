import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import { TitleSpan, RecommendedVideoCard, RecommendedVideoImg, RecommendedVideoText } from './RecommendedVideosComponent.styled';

const RecommendedVideos = ({ videos }) => {
  const recommendedVideos = videos.slice(0, 15);
  const { pathname } = useLocation();
  const currentPath = pathname.split('/');
  return (
    <>
      <div>
        {recommendedVideos.map((video) => (
          <Link
            to={
              currentPath[1] === 'video'
                ? `/video/${video.id.videoId}`
                : `/favorite-video/${video.id.videoId}`
            }
            style={{ textDecoration: 'none', color: 'var(--black)' }}
            key={video.etag}
          >
            <RecommendedVideoCard>
              <RecommendedVideoImg>
                <img
                  src={video.snippet.thumbnails.medium.url}
                  style={{ width: '100%' }}
                  alt={video.snippet.title}
                />
              </RecommendedVideoImg>
              <RecommendedVideoText >
                <TitleSpan>{video.snippet.title}</TitleSpan>
              </RecommendedVideoText>
            </RecommendedVideoCard>
            <hr style={{ width: '80%', margin: '0 auto', color: 'var(--main-blue)' }} />
          </Link>
        ))}
      </div>
    </>
  );
};

export default RecommendedVideos;
