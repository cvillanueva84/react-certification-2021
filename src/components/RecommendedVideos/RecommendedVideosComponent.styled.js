import styled from "styled-components";

export const TitleSpan = styled.span`
    font-size: 1.4rem;
    justify-content: left;
    width: 100%;
    display: inline-block;
`

TitleSpan.displayName = 'TitleSpan'

export const RecommendedVideoCard = styled.div`
  width: 100%;
  display: flex;
  margin-bottom: 1rem;
  margin-top: 1rem;
`;

RecommendedVideoCard.displayName = 'RecommendedVideoCard'

export const RecommendedVideoImg = styled.div`
  width: 40%;
  margin-left: 2rem;
`;
RecommendedVideoImg.displayName = 'RecommendedVideoImg'

export const RecommendedVideoText = styled.div`
  width: 60%;
  display: flex;
  align-items: center;
  font-size: 2rem;
`;

RecommendedVideoText.displayName = 'RecommendedVideoText'