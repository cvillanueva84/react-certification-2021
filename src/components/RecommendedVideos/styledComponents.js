import styled from "styled-components";

export const TitleSpan = styled.span`
    font-size: 2rem;
    justify-content: left;
    width: 100%;
    display: inline-block;
`