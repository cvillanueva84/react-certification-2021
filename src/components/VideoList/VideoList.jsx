import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { DivTitle, Container, Fakegrid } from './VideoList.styled';
import Card from '../Card/Card';

const baseURL =
  'https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json';

const VideoList = () => {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    async function getVideos() {
      const {
        data: { items },
      } = await axios.get(baseURL);
      const aux = [...items];
      aux.shift();
      setVideos(aux);
      // console.log(data.items.map((el) => el.snippet.description));
    }
    getVideos();
  }, []);

  return (
    <>
      <DivTitle>
        <h2>Welcome to the Challenge!</h2>
      </DivTitle>
      <Container>
        <Fakegrid>
          {videos.map((video) => (
            <Card
              title={video.snippet.title}
              description={video.snippet.description}
              link={video.snippet.thumbnails.high.url}
              key={video.snippet.publishedAt}
            />
          ))}
        </Fakegrid>
      </Container>
    </>
  );
};

export default VideoList;
