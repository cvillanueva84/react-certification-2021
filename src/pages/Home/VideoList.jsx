import React from 'react';
import styled from 'styled-components';
import VideoItem from './VideoItem';

const VideoListContainer = styled.div`
    width: 100%;
    height: 100vh;
    margin: 5px auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    
    & a {
        text-decoration: none;
    }
`;

const VideoList = ({videos}) => {

    if (videos.length < 0 ) return null;

    return (
        <VideoListContainer>
            {videos.items.map( (item) => (
                <VideoItem item={item} key={item.etag} />
            ))}
        </VideoListContainer>
    );

};

 export default VideoList;

 