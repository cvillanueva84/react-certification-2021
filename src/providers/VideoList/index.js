export {  VideoListProvider } from './VideoList.provider';
