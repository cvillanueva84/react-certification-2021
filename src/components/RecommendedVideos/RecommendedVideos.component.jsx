import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import './RecommendedVideos.styles.css';
import { TitleSpan } from './styledComponents';
const RecommendedVideos = () => {
  const { state } = useContext(VideoListContext);
  const { posts } = state;
  const recommendedVideos = posts.slice(0, 8);
  return (
    <>
      <div className="recommended-video__container">
        {recommendedVideos.map((video) => (
          <Link
            to={`/video/${video.id.videoId}`}
            style={{ textDecoration: 'none', color: '#000' }}
            key={video.id.videoId}
          >
            <div className="recommended-video__card">
              <div className="recommended-video__card-img">
                <img
                  src={video.snippet.thumbnails.medium.url}
                  style={{ width: '100%' }}
                  alt={video.snippet.title}
                />
              </div>
              <div className="recommended-video__card-text">
                <TitleSpan>{video.snippet.title}</TitleSpan>
              </div>
            </div>
          </Link>
        ))}
      </div>
    </>
  );
};

export default RecommendedVideos;
