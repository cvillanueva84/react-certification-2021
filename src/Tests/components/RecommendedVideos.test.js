import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import RecommendedVideos from '../../components/RecommendedVideos';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import { videosState } from '../mockData/videosList'


describe('Tests of the RecommendedVideos component', () => {
    const {posts} = videosState 

    const wrapper = mount(

        <VideoListContext.Provider
            value={{
                posts
            }}
            >
            <MemoryRouter>
                < RecommendedVideos videos={posts}/>
            </MemoryRouter>
        </VideoListContext.Provider>
    )

    

    test('should render the component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should display all of the videos passed', () => {
        let videos = wrapper.find('recommended-video__card')
        expect(videos.length).toBe(videos.length)
    })

})
