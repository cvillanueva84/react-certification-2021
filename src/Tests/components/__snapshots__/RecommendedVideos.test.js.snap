// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the RecommendedVideos component should render the component correctly 1`] = `
<div>
  <a
    href="/favorite-video/nmXMgqjQzls"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--black)",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="sc-gtsrHT iRpwMP"
    >
      <div
        className="sc-dlnjwi dIBHvn"
      >
        <img
          alt="Video Tour | Welcome to Wizeline Guadalajara"
          src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="sc-hKFxyN uEZHe"
      >
        <span
          className="sc-bdnxRM eNjdHH"
        >
          Video Tour | Welcome to Wizeline Guadalajara
        </span>
      </div>
    </div>
    <hr
      style={
        Object {
          "color": "var(--main-blue)",
          "margin": "0 auto",
          "width": "80%",
        }
      }
    />
  </a>
  <a
    href="/favorite-video/HYyRZiwBWc8"
    onClick={[Function]}
    style={
      Object {
        "color": "var(--black)",
        "textDecoration": "none",
      }
    }
  >
    <div
      className="sc-gtsrHT iRpwMP"
    >
      <div
        className="sc-dlnjwi dIBHvn"
      >
        <img
          alt="Wizeline Guadalajara | Bringing Silicon Valley to Mexico"
          src="https://i.ytimg.com/vi/HYyRZiwBWc8/mqdefault.jpg"
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="sc-hKFxyN uEZHe"
      >
        <span
          className="sc-bdnxRM eNjdHH"
        >
          Wizeline Guadalajara | Bringing Silicon Valley to Mexico
        </span>
      </div>
    </div>
    <hr
      style={
        Object {
          "color": "var(--main-blue)",
          "margin": "0 auto",
          "width": "80%",
        }
      }
    />
  </a>
</div>
`;
